Initial osx implementation

chemecse opened the pull request:
This is an initial port of Papaya to OSX.

Caveats:
- Since EasyTab doesn't support OSX I added a `PAPAYA_NO_EASYTAB_SUPPORT` define
- Window resizing is currently disabled because glViewport doesn't behave normally with NSOpenGLView

chemecse committed:
Initial osx implementation

0xi commented:
trying to build this i got two errors:
/usr/include/mach/mach_host.h has got 2 structs with fields named "internal" which the macro wants to change to "static".
i dont know what the mach header does, so i just changed all the internal's to static by hand in order to build fine.

chemecse commented:
@0xi Hey! Thanks for testing :). If you are okay with it, I would ask that you post the exact output from running `make`, the output from running `g++ -v`, and your version of OSX.
Thanks again.

chemecse committed:
Reorder include files in papaya_platform_osx.mm

Should fix an issue with regards to the internal
define overwriting a data structure in mach

chemecse commented:
@0xi I reordered the includes so the internal define wouldn't interfere with the mach headers. Let me know if this fixes the issue you were having.

0xi commented:
reordering fixed this issue! builds fine now!
thank you :)

EDIT
if you are still interested in the error:

/usr/include/mach/mach_host.h:97:12: error: invalid storage class specifier in
      function declarator
        boolean_t internal,
../../src/libs/types.h:6:18: note: expanded from macro 'internal'
#define internal static

for line 97 and line 219, running El Capitan 10.11.4

EDIT 2
g++ -v output
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 7.3.0 (clang-703.0.29)
Target: x86_64-apple-darwin15.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

ApoorvaJ commented:
@chemecse Based on our discussion, I have manually integrated your platform-agnostic changes into this repo, and I have posted a link to your fork on the front page. Is it okay if I close this pull request?

chemecse commented:
@ApoorvaJ Meant to do this yesterday, closing now!
